components/movementsensor/gpsutils/cachedData.go

```diff
@@ -224,11 +231,15 @@ func (g *CachedData) ReadSatsInView(ctx context.Context) (int, error) {
 	return g.nmeaData.SatsInView, nil
 }
 
-// ReadSatsInUse returns the number of satellites in use to calculate fix type.
-func (g *CachedData) ReadSatsInUse(ctx context.Context) (int, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	return g.nmeaData.SatsInUse, nil
+// GetCommonReadings returns a map including fix value, sats in view and sats in use.
+func (g *CachedData) GetCommonReadings(ctx context.Context) (map[int]interface{}, error) {
+	commonReadings := make(map[int]interface{})
```

Comment:
You construct a map here, but every time this is used, you take things back out of the map for use elsewhere. Why not just have this function return a tuple?
```go
 func (g *CachedData) GetCommonReadings(ctx context.Context) (int, int, int) {
   ...
}
```

If the answer is that it's hard to remember which index in the tuple is which, consider making it a struct instead of a map:
```go
type CommonReadings struct {
    FixValue, SatsInView, SatsInUse int
}

func (g *CachedData) GetCommonReadings(ctx context.Context) CommonReadings {
    ...
}
```
(What I'm really trying to avoid is all the extra complexity that comes with constructing a map that could contain arbitrary keys, when you'd have both faster code and better type safety if instead of arbitrary keys you only had exactly the things you were expecting to have.)

Reply:
I like the struct idea. Implementing that!